ckan/lib/search/query.py

```diff
@@ -310,18 +315,23 @@ def run(self, query):
             rows_to_query = rows_to_return
         query['rows'] = rows_to_query
 
+        fq = []
+        if 'fq' in query:
+            fq.append(query['fq'])
+        fq.extend(query.get('fq_list', []))
+
         # show only results from this CKAN instance
-        fq = query.get('fq', '')
-        if not '+site_id:' in fq:
-            fq += ' +site_id:"%s"' % config.get('ckan.site_id')
+        fq.append('+site_id:%s' % solr_literal(config.get('ckan.site_id')))
 
         # filter for package status
-        if not '+state:' in fq:
-            fq += " +state:active"
-        query['fq'] = [fq]
+        if not '+state:' in query.get('fq', ''):
+            fq.append('+state:active')
 
-        fq_list = query.get('fq_list', [])
-        query['fq'].extend(fq_list)
+        # only return things we should be able to see
+        if permission_labels is not None:
+            fq.append('+permission_labels:(%s)' % ' OR '.join(
```

Comment:
I decided to clearly separate permissions-related information from user metadata. The only place to set permission labels is the IPermissionLabels interface.